chore(docs): Update 'CONTRIBUTING.md' to include info about syncing the fork

Yogita98 opened the pull request:
## Because

- The CONTRIBUTING.md file doesn't include instructions on how to sync the fork with the original repository.

## This pull request

- Aims to include the information and steps needed to sync the local fork with the original repository 
which can be very helpful for first-time contributers

## Issue that this pull request solves

-

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

Yogita98 commented:
@LZoog @jodyheavener I have created this PR which intends to include information on syncing the local fork with the original repository. I'm not sure if it was even needed but thought that it might come as handy for first-time contributors.

LZoog commented on CONTRIBUTING.md:
```diff
@@ -127,6 +127,25 @@ reference GitHub issues that this commit **Closes**. In order to maintain a refe
+## What should you do once your PR is merged?
+
+Once your PR is merged, it's recommended to pull new changes from the `main` (which will also include the one that you made).
+So, for that purpose specify a new remote `upstream` repository which will be pointing to the actual [mozilla/fxa](https://github.com/mozilla/fxa) repo.
```
I'm not sure if I see a note anywhere about contributors pushing to and PR'ing from their fork. FxA team members (part of the Mozilla organization) can push upstream, but contributors must push to their fork and put a PR in from their fork. It might be worth explicitly calling that out in another section? 🙂

Also, I think this can maybe be very slightly rephrased (around "new remote `upstream` repository"), since I'd still consider it the same repository, just a different remote. Maybe just lose the word "repository" so it reads "...specify a new remote `upstream` which will..."?

Yogita98 replied:
Ohhh my bad that it came up like that. 
I created this PR just with the intention of setting a new remote to fetch new updates from. I could close it right away if it's not needed?
Yeah, it completely makes sense to remove the word repository.

LZoog commented on CONTRIBUTING.md:
```diff
@@ -127,6 +127,25 @@ reference GitHub issues that this commit **Closes**. In order to maintain a refe
+Once your PR is merged, it's recommended to pull new changes from the `main` (which will also include the one that you made).
+So, for that purpose specify a new remote `upstream` repository which will be pointing to the actual [mozilla/fxa](https://github.com/mozilla/fxa) repo.
+
+git remote add upstream https://github.com/mozilla/fxa.git
```
Can we add backticks around this too?

Yogita98 replied:
Yeah sure! If we're keeping this change I will commit the required changes.

LZoog reviewed (commented):
Thank you for writing this! Anything that helps make something more clear for contributors is a welcomed addition to our docs. I just have a couple of suggestions. 🙂

jodyheavener commented on CONTRIBUTING.md:
```diff
@@ -127,6 +127,25 @@ reference GitHub issues that this commit **Closes**. In order to maintain a refe
 `fixes #<issue_number>` if it closes a related issue or `issue #<issue_number>`
 if it's a partial fix.
 
+## What should you do once your PR is merged?
```
I might almost re-contextualize this section to be more broadly about syncing your fork; there are plenty of reasons to keep your fork in sync beyond your PR getting merged. What do you think? I am so happy you documented this though 🙂

Yogita98 replied:
Yeah right, since I opened this PR simply with a view to help out new contributors, I most certainly didn't realize the scope of keeping the fork in sync. Thanks for pointing that out! 
With this point being made, I wonder if this section after re-contextualizing should be kept under CONTRIBUTING.md or README.md? What are your thoughts? Also, thanks a lot for the acknowledgement :)

jodyheavener replied:
I think it's probably still fine in the Contributing doc since it deals with Git and your setup more-so than working on the code.

Yogita98 replied:
> I think it's probably still fine in the Contributing doc since it deals with Git and your setup more-so than working on the code.

Yeah makes sense!

Yogita98 commented:
I have updated my PR and
1. Recontextualized the section in order to broaden its scope
2. Added backticks wherever missing
3. The last commit was missing instructions about syncing the fork on Github. It only described the steps needed to keep the local copy in sync so this commit intends to add the steps needed to keep the fork updated on GitHub.

LZoog commented:
@Yogita98 thanks so much, this looks great! I might let Jody comment back here too in case he has any other feedback, but will you squash your commits into one (I always run `git rebase -i main`)? 🙂

Yogita98 commented:
@LZoog Squashed the commits into one as suggested by you. What do you think @jodyheavener ?

jodyheavener reviewed (approved):
Fantastic work, @Yogita98!

jodyheavener commented:
Oh, whoops, looks like your squashed commit isn't signed @Yogita98 - sorry, there's so much to consider for our process!

Yogita98 commented:
No problem @jodyheavener . I will sign it right away

Yogita98 commented:
Hey @jodyheavener 
I tried to sign the commit by amending the commit but for some reason, it's not getting reflected here. I can see it signed in my local though. The following are the steps that I followed:
```
git rebase -i HEAD~1
git commit --amend --no-edit -S
git push --force-with-lease origin my-branch
```
 Can you tell me if I'm doing it the right way?

jodyheavener commented:
@Yogita98 huh that's odd. If it's showing locally but not here, do you have your GPG key set up with your [GitHub profile](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-gpg-key-to-your-github-account)? You can always undo the commit entirely (`git reset --soft HEAD~1`), and then re-commit with signing enabled.

Yogita98 commented:
Thanks alot @jodyheavener! Apologize for the mess though

jodyheavener reviewed (approved):
Awesome!
